0000-interledger-tokens/0000-interledger-tokens.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Interledger Tokens
+draft: 1
+---
+# Interledger Tokens
+
+An **Interledger Token** is a special form of JSON Web Token that can be used to initiate an Interledger payment. The token provides proof that the payment has been pre-authorized by the payer. By using these tokens new use cases for "debit pull" style payments such as recurring subscriptions or offline authorization are possible.
+
+## Background
+
+TODO
+
+## Motivation
+
+TODO
+
+## Terminology
+
+The following terms are used throughout this specification. Where they are used their meaning as follows:
+
+### Payer
+
+The entity whose funds are being spent when the token is used to initiate a payment.
+
+### Payer's Interledger Service Provider (ILSP)
+
+The entity that accepts and processes the token and payment request. This may be the payer or they may delegate it to some other entity that is able to make payments on their behalf.
+
+### Payee
+
+The entity that is requesting payment using the token.
+
+## The Interledger Token
+
+An **Interledger Token** is a special form of a JSON Web Token (JWT) [RFC7519](https://tools.ietf.org/html/rfc7519) that can be submitted to a participant on the Interledger network to request that they initiate a new payment via ILP. The content of the token describes how much to pay, who to pay and provides information for the entity processing the token to verify that the payment was authorized by the owner of the funds being sent.
+
+**Interledger Tokens** are limited profile of a standard JWT. This profile limits the allowed algorithms, allowed claims and defines specific rules for what constitutes valid claim values. A token is used by a payee to "claim" that a payment has been authorized by the payer in accordance with the terms of the token.
+
+After processing a token and making a payment the participant that has made the payment MAY return a new token to the payee to use again in the future. For example, a payee may posses a token that authorizes a payment to them of USD 100. They may then submit this to initiate a payment but request that only USD 25 is sent in the payment. Following successful execution of the payment they may be returned a new token that authorizes a payment to them of USD 75.
+
+An example token payload is shown below (with further details provided in later sections):
+
+```json
+{
+  "jti": "48359d89-e9ef-44dd-9a3d-68f991ed7556", //Unique Token ID
+  "iss": "872369652347412343", //Unique identifier for the payer (recognized by their ILSP)
+  "sub": "example.payee123", //ILP Address of the payee
```

Comment:
Putting the address without the secret isn't that useful because if someone is able to give you a bad secret to do PSK with, they could re-route the money to themselves and provide valid fulfillments even if it didn't get to the real ILP address.

Reply:
It's a bearer token. That's okay.

Reply:
Also, if ILP addresses are ephemeral, your receiver might be listening on a different address by the time the merchant gets around to paying you

Reply:
I'd assume as a receiver you'd have some control over that or at least get short lived tokens if you expect your address to change often.

Bare in mind that the majority of payments are one off. The payee will get the token from the payer, redeem it and then move on all within probably a few seconds.